src/plugins/image_embeddable/public/components/image_embeddable.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useState } from 'react';
+import useMount from 'react-use/lib/useMount';
+
+import { ReactEmbeddableDynamicActionsApi } from '@kbn/embeddable-enhanced-plugin/public/plugin';
+import { PublishingSubject, useStateFromPublishingSubject } from '@kbn/presentation-publishing';
+
+import { imageClickTrigger } from '../actions';
+import { ImageEmbeddableApi } from '../image_embeddable/types';
+import { FileImageMetadata, FilesClient, imageEmbeddableFileKind } from '../imports';
+import { coreServices, screenshotModeService, uiActionsService } from '../services/kibana_services';
+import { ImageConfig } from '../types';
+import { createValidateUrl } from '../utils/validate_url';
+import { ImageViewer } from './image_viewer';
+import { ImageViewerContext } from './image_viewer/image_viewer_context';
+
+import './image_embeddable.scss';
+
+interface ImageEmbeddableProps {
+  api: ImageEmbeddableApi & {
+    setDataLoading: (loading: boolean | undefined) => void;
+    imageConfig$: PublishingSubject<ImageConfig>;
+  };
+  filesClient: FilesClient<FileImageMetadata>;
+  startDynamicActions?: ReactEmbeddableDynamicActionsApi['startDynamicActions'];
+}
+
+export const ImageEmbeddable = ({
+  api,
+  filesClient,
+  startDynamicActions,
+}: ImageEmbeddableProps) => {
+  const imageConfig = useStateFromPublishingSubject(api.imageConfig$);
+  const dynamicActionsState = useStateFromPublishingSubject(api.dynamicActionsState$);
+
+  const [hasTriggerActions, setHasTriggerActions] = useState(false);
+
+  useMount(() => {
+    api.setDataLoading(true);
+  });
+
+  useEffect(() => {
+    if (!startDynamicActions) return;
+    // start the dynamic actions manager on mount
+    const { stopDynamicActions } = startDynamicActions();
```

Comment:
I get what you're saying - unfortunately, `stopDynamicActions` **must** currently be called when the component gets unmounted or the drilldowns get duplicated as you navigate around. And I don't think there is a way to handle this cleanup functionality from within `buildEmbeddable`? So I'm not sure how else to do it ATM - I think this might just be a limitation of how the dynamic actions manager currently works - as we already know, this code desperately needs some work 🙈

Reply:
I think it's a pretty common pattern to have a service that needs to be started on mount and stopped on unmount. Our examples generally just put a `useEffect` cleanup function right in the component in the factory just for cleanup, but that's _essentially_ what you're doing here.

Maybe we do need an explicit `onDestroy` event, which could be passed in during `buildEmbeddable` and called when the renderer gets unmounted.

Reply:
That's fair! I can move it up a level if that is the pattern we want to establish :+1: And yes, then I could call the `start` in `buildEmbeddable` rather than in a `useEffect`.

Reply:
Done in 61dd6dfa133ee1c04256c493f773cdac2176b0bb